File: Programmers/Lv.1/두개뽑아더하기.py
def solution(numbers):
    answer = []
    
    for i in range(len(numbers) - 1):
        for j in range(1, len(numbers)):
            if i != j:
                answer.append(numbers[i] + numbers[j])
    
    answer = list(set(answer))
    answer.sort()
    
    return answer

# def solution(numbers):
#     answer = []
#     for i in range(len(numbers)):
#         for j in range(i + 1, len(numbers)):
#             answer.append(numbers[i] + numbers[j])
#     return sorted(list(set(answer)))

File: Programmers/Lv.1.Kotlin/두개뽑아더하기.kt
class Solution {
    fun solution(numbers: IntArray): IntArray {
        var answer = arrayListOf<Int>()
        
        for (i in 0 until (numbers.size - 1)) {
            for (j in 1 until numbers.size) {
                if (i != j) {
                    answer.add(numbers[i] + numbers[j])
                }
            }
        }
        
        answer.sort()
        return answer.distinct().toIntArray()

        // val list = numbers.toList()

        // return list.withIndex().flatMap { i -> list.withIndex().map { j -> i to j } }
        //     .filter { it.first.index != it.second.index }
        //     .map { it.first.value + it.second.value }
        //     .toSortedSet()
        //     .toIntArray()
    }
}